scanner/src/test/java/ca/on/oicr/gsi/runscanner/scanner/processor/AbstractProcessorTest.java

```diff
@@ -29,7 +31,24 @@ protected final void checkDirectory(String root)
           null); // We delete this because it is going to change during updates to dependencies.
       result.setSequencerFolderPath(
           null); // We delete this because it is going to be different in each environment.
-      result.setMetrics(null); // We delete these because changes in metrics are non-critical.
+
+      // For all non-Illumina processors
+      if (!clazz.equals(IlluminaNotificationDto.class)) {
+        result.setMetrics(null); // We delete these because changes in metrics are non-critical.
+      }
+
+      // For Illumina processors only
+      if (clazz.equals(IlluminaNotificationDto.class)) {
+        // Load the metrics into a Jackson JsonNode to test for equality
+        ObjectMapper MapperTest = new ObjectMapper();
+        JsonNode jsonNodeResult = MapperTest.readTree(result.getMetrics());
+        JsonNode jsonNodeReference = MapperTest.readTree(reference.getMetrics());
+        if (jsonNodeResult.equals(jsonNodeReference)) {
+          // If Json objects are equal, set string representation to also be equal
+          reference.setMetrics(result.getMetrics());
```

Comment:
rather than manipulate the result, fail the test early if `!jsonNodeResult.equals(jsonNodeReference)` then `result.setMetrics(null)` in all remaining cases (ie, after the if block entirely)